sw/device/lib/dif/README.md

```diff
@@ -50,6 +51,30 @@ there are some relaxations of these rules for them described at the end.
 * DIFs must use enum return codes rather than booleans for reporting errors. If
   a DIF can either error or instead produce a value, it should return an enum
   return code, and use an out-parameter for returning the produced value.
+  * DIF libraries must have a general return code enum called
+    `dif_<ip>_result_t`. It must define the following constants (in this order):
+    * `kDif<ip>Ok`, with value 0, to denote the call succeeded.
+    * `kDif<ip>Error`, to denote a non-specific error happened during the call.
```

Comment:
Is `Error` the best term here? After all `BadArg` is also an error. Maybe something like `UnknownError` or just `GeneralError`? Or is the goal here to decrease verbosity?

Reply:
I think `kDif<ip>Error` is fine, we discussed it #1911 . In my opinion all of these variants are fine, so we might just keep the present one.

Reply:
While we are at it, maybe we should remove individual `dif_<ip>_result_t`s and define a common `dif_result_t`. What do you think?

Reply:
The idea of a single `dif_result_t` was floated at the call, and I am very against it, because it would have to live in a centralised header, and I do not want more dependencies for the DIFs. 

We're trying to ensure the DIFs remain reusable on other projects and so can be individually vendored like the peripheral RTL can be vendored. We cannot really avoid the use of the `mmio` library (and the like), but I want to ensure we keep other dependencies to a minimum. I don't see the payoff for sharing an enum like this yet.

Reply:
I think it is prudent to give explicit values to the `dif_<ip>_result_t` constants, and any in more specialised enums (so that the `Ok`, `Error`, and `BadArg` values match).

Reply:
I think the "main" payoff is being able to write something like `RETURN_IF_ERR` or like Rust's `?` operator. However, it is still unclear if this has that much utility, and we can simply mandate that `Ok` always has the representation `0`.